src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -86,6 +88,29 @@ public static bool ZipDeploymentAppSettingsExist(this IEnvironment environment)
                    !string.IsNullOrEmpty(environment.GetEnvironmentVariable(ScmRunFromPackage));
         }
 
+        public static bool AzureFilesAppSettingsExist(this IEnvironment environment)
+        {
+            return !string.IsNullOrEmpty(environment.GetEnvironmentVariable(AzureFilesConnectionString)) &&
+                   !string.IsNullOrEmpty(environment.GetEnvironmentVariable(AzureFilesContentShare));
+        }
+
+        public static FunctionAppContentEditingState IsFunctionAppContentEditable(this IEnvironment environment, IOptions<ScriptApplicationHostOptions> applicationHostOptions)
```

Comment:
Please add unit tests for these 2 new methods in EnvironmentExtensionsTests.

With that, IsFunctionAppContentEditable (and rename since this is no longer returning bool) could be a private method in HostController or a public method in some utility class.
currently it seems this method is not just looking up things from IEnvironment unlike the rest of the methods in this class, so feels out of place here.

Reply:
Moved to Utility.